manager/integration/tests/backupstore.py

```diff
@@ -16,13 +16,102 @@
 from common import is_backupTarget_nfs
 from common import get_longhorn_api_client
 from common import delete_backup_volume
+from common import get_backupstore_url
 
 BACKUPSTORE_BV_PREFIX = "/backupstore/volumes/"
 
-
 TEMP_FILE_PATH = "/tmp/temp_file"
 
 
+@pytest.fixture
+def backupstore_s3(client):
+    set_backupstore_s3(client)
+    yield
+    reset_backupstore_setting(client)
+
+
+@pytest.fixture
+def backupstore_nfs(client):
+    set_backupstore_nfs(client)
+    yield
+    reset_backupstore_setting(client)
+
+
+@pytest.fixture(params=["s3", "nfs"])
+def set_random_backupstore(request, client):
+    if request.param == "s3":
+        set_backupstore_s3(client)
+    elif request.param == "nfs":
+        set_backupstore_nfs(client)
+        mount_nfs_backupstore(client)
+
+    yield
+    reset_backupstore_setting(client)
```

Comment:
Consider running cleanup_backupstore before resetting the settings.
This way the fixture is responsible for cleaning up behind itself.
This will also then differentiate between a test failure or a failure in the fixture.

Reply:
fixed